src/nvim/tui/tui.c

```diff
@@ -44,19 +44,17 @@
 #define OUTBUF_SIZE 0xffff
 
 #define TOO_MANY_EVENTS 1000000
-#define STARTS_WITH(str, prefix) (!memcmp(str, prefix, sizeof(prefix) - 1))
-#define TMUX_WRAP(seq) (is_tmux ? "\x1bPtmux;\x1b" seq "\x1b\\" : seq)
-
-typedef enum TermType {
-  kTermUnknown,
-  kTermGnome,
-  kTermiTerm,
-  kTermKonsole,
-  kTermRxvt,
-  kTermDTTerm,
-  kTermXTerm,
-  kTermTeraTerm,
-} TermType;
+#define STARTS_WITH(str, prefix) (strlen(term) >= (sizeof(prefix) - 1) \
+    && 0 == memcmp((str), (prefix), sizeof(prefix) - 1))
+#define TMUX_WRAP(is_tmux,seq) ((is_tmux) ? "\x1bPtmux;\x1b" seq "\x1b\\" : seq)
+#define LINUXRESETC "\x1b[?0c"
+
+// Per the commentary in terminfo, only a minus sign is a true suffix
+// separator.
+#define TERMINAL_FAMILY(term, prefix) ((term) \
+    && strlen(term) >= (sizeof(prefix) - 1) \
+    && 0 == memcmp((term), (prefix), sizeof(prefix) - 1) \
+    && ('\0' == (term)[sizeof(prefix) - 1] || '-' == (term)[sizeof(prefix) - 1]))
```

Comment:
True.  Of course that requires common subexpression analysis into functions.  Not that we are particularly worried about optimizing it.  (-: